doc/Language/phasers.pod

```diff
@@ -0,0 +1,357 @@
+=begin pod
+
+=TITLE Phasers
+
+=SUBTITLE Execution phases
+
+The lifetime (execution timeline) of a program is broken up into phases. A
+I<phaser> is a block of code called during a specific execution phase.
+
+=head1 Phasers
+
+A phaser block is just a trait of the closure containing it, and is
+automatically called at the appropriate moment. These auto-called blocks are
+known as I<phasers>, since they generally mark the transition from one phase of
+computing to another. For instance, a C<CHECK> block is called at the end of
+compiling a compilation unit. Other kinds of phasers can be installed as well;
+these are automatically called at various times as appropriate, and some of
+them respond to various control exceptions and exit values.
+
+Here is a summary:
+
+      BEGIN {...} #  * at compile time, ASAP, only ever runs once
+      CHECK {...} #  * at compile time, ALAP, only ever runs once
+       LINK {...} #  * at link time, ALAP, only ever runs once
+       INIT {...} #  * at run time, ASAP, only ever runs once
+        END {...} #  at run time, ALAP, only ever runs once
+
+      ENTER {...} #  * at every block entry time, repeats on loop blocks.
+      LEAVE {...} #  at every block exit time (even stack unwinds from exceptions)
```

Comment:
Later it goes into some info on this -- I think it means that if something way outside catches an exception, AFTER the exception is done it will run all of the intermediate LEAVE phasers. If, however, you cancel the exception (I don't even know how to do that), then the original execution point isn't completed, so the stacks are not popped and execution continues without calling LEAVE until later.